include/benchmark/benchmark_api.h

```diff
@@ -230,7 +230,7 @@ BENCHMARK_UNUSED static int stream_init_anchor = InitializeStreams();
 // expression from being optimized away by the compiler. This function is
 // intended to add little to no overhead.
 // See: https://youtu.be/nXaxk27zwlk?t=2441
-#if defined(__GNUC__)
+#if defined(__GNUC__) && !defined(__pnacl__) && !defined(EMSCRIPTEN)
```

Comment:
I checked that the current implementation of `benchmark::DoNotOptimize` results in *seemingly* reasonable numbers on PNaCl and Asm.js, but it would certainly help to have documented built-in support for this idiom in compilers.